src/mgmt/tests/srv_drpc_tests.c

```diff
@@ -1034,6 +1055,126 @@ test_drpc_pool_list_cont_with_containers(void **state)
+	size_t		i;
+
+	/* Fill out with junk that won't translate to a PoolSetPropReq */
+	for (i = 0; i < sizeof(bad_bytes); i++)
+		bad_bytes[i] = i;
+
+	call.body.data = bad_bytes;
+	call.body.len = sizeof(bad_bytes);
+
+	ds_mgmt_drpc_pool_set_prop(&call, &resp);
+
+	assert_int_equal(resp.status, DRPC__STATUS__FAILED_UNMARSHAL_PAYLOAD);
+
+	assert_null(resp.body.data);
+	assert_int_equal(resp.body.len, 0);
+}
+
+static void
+setup_pool_set_prop_drpc_call(Drpc__Call *call, Mgmt__PoolSetPropReq *req)
+{
+	size_t			len;
+	uint8_t			*body;
+
+	len = mgmt__pool_set_prop_req__get_packed_size(req);
+	D_ALLOC(body, len);
+	assert_non_null(body);
+
+	mgmt__pool_set_prop_req__pack(req, body);
+
+	call->body.data = body;
+	call->body.len = len;
+}
+
+static void
+expect_drpc_pool_set_prop_resp_with_error(Drpc__Response *resp,
+					int expected_err)
+{
+	Mgmt__PoolSetPropResp *set_prop_resp = NULL;
+
+	assert_int_equal(resp->status, DRPC__STATUS__SUCCESS);
+	assert_non_null(resp->body.data);
+
+	set_prop_resp = mgmt__pool_set_prop_resp__unpack(NULL,
+			resp->body.len, resp->body.data);
+	assert_non_null(set_prop_resp);
+	assert_int_equal(set_prop_resp->status, expected_err);
+
+	mgmt__pool_set_prop_resp__free_unpacked(set_prop_resp, NULL);
+}
+
+static void
+test_drpc_pool_set_prop_invalid_property_type(void **state)
+{
+	Drpc__Call call = DRPC__CALL__INIT;
+	Drpc__Response resp = DRPC__RESPONSE__INIT;
+	Mgmt__PoolSetPropReq req = MGMT__POOL_SET_PROP_REQ__INIT;
+
+	req.uuid = TEST_UUID;
+	/* make the value valid to ensure we're testing the property */
+	req.numval = 1;
+	req.value_case = MGMT__POOL_SET_PROP_REQ__VALUE_NUMVAL;
+	setup_pool_set_prop_drpc_call(&call, &req);
+
+	ds_mgmt_drpc_pool_set_prop(&call, &resp);
+
+	expect_drpc_pool_set_prop_resp_with_error(&resp, -DER_INVAL);
+
+	D_FREE(call.body.data);
+	D_FREE(resp.body.data);
+}
+
+static void
+test_drpc_pool_set_prop_invalid_value_type(void **state)
+{
+	Drpc__Call call = DRPC__CALL__INIT;
+	Drpc__Response resp = DRPC__RESPONSE__INIT;
+	Mgmt__PoolSetPropReq req = MGMT__POOL_SET_PROP_REQ__INIT;
+
+	req.uuid = TEST_UUID;
+	req.number = 1; /* doesn't matter */
+	req.property_case = MGMT__POOL_SET_PROP_REQ__PROPERTY_NUMBER;
+	setup_pool_set_prop_drpc_call(&call, &req);
+
+	ds_mgmt_drpc_pool_set_prop(&call, &resp);
+
+	expect_drpc_pool_set_prop_resp_with_error(&resp, -DER_INVAL);
+
+	D_FREE(call.body.data);
+	D_FREE(resp.body.data);
+}
+
+static void
+test_drpc_pool_set_prop_bad_uuid(void **state)
+{
+	Drpc__Call call = DRPC__CALL__INIT;
+	Drpc__Response resp = DRPC__RESPONSE__INIT;
+	Mgmt__PoolSetPropReq req = MGMT__POOL_SET_PROP_REQ__INIT;
+
+	req.uuid = "wow this won't work";
+	req.number = 1; /* doesn't matter */
+	req.property_case = MGMT__POOL_SET_PROP_REQ__PROPERTY_NUMBER;
+	setup_pool_set_prop_drpc_call(&call, &req);
+
+	ds_mgmt_drpc_pool_set_prop(&call, &resp);
+
+	expect_drpc_pool_set_prop_resp_with_error(&resp, -DER_INVAL);
+
+	D_FREE(call.body.data);
+	D_FREE(resp.body.data);
+}
+
```

Comment:
Should test the positive (success) case as well.

Reply:
Done